src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.Integer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encoding.Tests.Cbor
+{
+    public partial class CborReaderTests
+    {
+        // Data points taken from https://tools.ietf.org/html/rfc7049#appendix-A
+
+        [Theory]
+        [InlineData(0, "00")]
+        [InlineData(1, "01")]
+        [InlineData(10, "0a")]
+        [InlineData(23, "17")]
+        [InlineData(24, "1818")]
+        [InlineData(25, "1819")]
+        [InlineData(100, "1864")]
+        [InlineData(1000, "1903e8")]
+        [InlineData(1000000, "1a000f4240")]
+        [InlineData(1000000000000, "1b000000e8d4a51000")]
+        [InlineData(-1, "20")]
+        [InlineData(-10, "29")]
+        [InlineData(-100, "3863")]
+        [InlineData(-1000, "3903e7")]
+        public static void Int64ReaderTests(long expectedResult, string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborValueReader(data);
+            long actualResult = reader.ReadInt64();
+            Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Theory]
+        [InlineData(0, "00")]
+        [InlineData(1, "01")]
+        [InlineData(10, "0a")]
+        [InlineData(23, "17")]
+        [InlineData(24, "1818")]
+        [InlineData(25, "1819")]
+        [InlineData(100, "1864")]
+        [InlineData(1000, "1903e8")]
+        [InlineData(1000000, "1a000f4240")]
+        [InlineData(1000000000000, "1b000000e8d4a51000")]
+        [InlineData(18446744073709551615, "1bffffffffffffffff")]
+        public static void UInt64ReaderTests(ulong expectedResult, string hexEncoding)
+        {
+            byte[] data = hexEncoding.HexToByteArray();
+            var reader = new CborValueReader(data);
+            ulong actualResult = reader.ReadUInt64();
+            Assert.Equal(expectedResult, actualResult);
+        }
+    }
```

Comment:
There are a lot of exceptions that aren't being tested for.  Not failing for indefinite is fine, but major type 0 or 1 with an additional information of 30 is a spec violation, so it should be tested for.

Reply:
I was planning on adding those later on, when the shape of the implementation has become clearer. Are you sure you want me to work on those now?

Reply:
Yep. The 3-5 days per checklist item is so that every stage PR has 100% code coverage and feature coverage.  (Modulo Debug.Fail blocks and NotSupportedExceptions for things deferred to later stages)

Reply:
Ok.

Reply:
I'd probably also need to add Int32 and UInt32 overloads as well in this iteration.